src/mapgen/mg_ore.cpp

```diff
@@ -371,17 +371,25 @@ void OreVein::generate(MMVManip *vm, int mapseed, u32 blockseed,
 	PcgRandom pr(blockseed + 520);
 	MapNode n_ore(c_ore, 0, ore_param2);
 
-	u32 sizex = (nmax.X - nmin.X + 1);
-
-	if (!noise) {
-		int sx = nmax.X - nmin.X + 1;
-		int sy = nmax.Y - nmin.Y + 1;
-		int sz = nmax.Z - nmin.Z + 1;
-		noise  = new Noise(&np, mapseed, sx, sy, sz);
-		noise2 = new Noise(&np, mapseed + 436, sx, sy, sz);
+	int sizex = nmax.X - nmin.X + 1;
+	int sizey = nmax.Y - nmin.Y + 1;
+	// Because this ore uses 3D noise the perlinmap Y size can be different in
+	// different mapchunks due to ore Y limits. So recreate the noise objects
+	// if Y size has changed.
+	// Because these noise objects are created multiple times for this ore type
+	// it is necessary to 'delete' them here.
+	if (!noise || sizey != sizey_prev) {
+		if (noise)
```

Comment:
useless if

Reply:
?

Reply:
Oh you mean just 'delete'? The noise has to exist if we delete (i think).

Reply:
Maybe not, they are initialised as nullptr, will try.

Reply:
Done.